workflows/phylorthology.nf

```diff
@@ -97,280 +85,204 @@ def create_og_channel(LinkedHashMap row) {
-    //     "broad",
-    //     [],
-    //     []
-    //     )
-        
+    ORTHOFINDER_PREP(complete_prots_list, "complete_dataset")
+    ORTHOFINDER_PREP_TEST(mcl_test_prots_list, "mcl_test_dataset")
+    ch_versions = ch_versions.mix(ORTHOFINDER_PREP.out.versions)
+
     //
     // MODULE: All-v-All diamond/blastp
     //
     // Run for the test set (used to determine the best value of the MCL
     // inflation parameter)
-    ch_blastp_mcl_test = DIAMOND_BLASTP_TEST (
-        ch_orthof_mcl_test,
-        ch_test_dmd,
+    DIAMOND_BLASTP_TEST(
+        ch_all_data.mcl_test_prots,
+        ORTHOFINDER_PREP_TEST.out.fastas.flatten(),
+        ORTHOFINDER_PREP_TEST.out.diamonds.flatten(),
         "txt",
         "true",
         []
     )
-    .txt
-    
-    // And for the full dataset, to be clustered into orthogroups using 
-    // the best inflation parameter. 
-    ch_blastp_complete = DIAMOND_BLASTP (
-        ch_orthof_complete,
-        ch_dmd,
+
+    // And for the full dataset, to be clustered into orthogroups using
+    // the best inflation parameter.
+    DIAMOND_BLASTP(
+        ch_all_data.complete_prots,
+        ORTHOFINDER_PREP.out.fastas.flatten(),
+        ORTHOFINDER_PREP.out.diamonds.flatten(),
         "txt",
         "false",
         []
     )
-    .txt
     ch_versions = ch_versions.mix(DIAMOND_BLASTP.out.versions)
 
     //
     // MODULE: Run Orthofinder's implementation of MCL (with similarity score
     //         correction).
     //
-    // Collect all pairwise similarity scores into a single channel and pass to
-    // the orthofinder MCL analysis so that it doesn't start until the full 
-    // set of all-v-all comparisons have completed.
-    ch_simil_mcl_test = ch_blastp_mcl_test.mix(ch_blastp_mcl_test).collect()
-    ch_simil_complete = ch_blastp_complete.mix(ch_blastp_complete).collect()
-
-    // First determine the optimal MCL inflation parameter, and then 
-    // subsequently use this for full orthogroup inference. 
-    ch_mcl = ORTHOFINDER_MCL_TEST (
+
+    // TODO: Fix the output_dir determination logic
+    // First determine the optimal MCL inflation parameter, and then
+    // subsequently use this for full orthogroup inference.
+    ORTHOFINDER_MCL_TEST(
         ch_inflation,
-        ch_simil_mcl_test,
-        "true"
+        DIAMOND_BLASTP_TEST.out.txt.collect(),
+        ORTHOFINDER_PREP_TEST.out.fastas,
+        ORTHOFINDER_PREP_TEST.out.diamonds,
+        ORTHOFINDER_PREP_TEST.out.sppIDs,
+        ORTHOFINDER_PREP_TEST.out.seqIDs,
+        "mcl_test_dataset"
     )
-    .og_fpath
-    
+
     //
     // MODULE: COGEQC
     // Run an R-script that applies cogqc to assess orthogroup inference
     // accuracy/performance.
     //
-    ch_cogeqc = COGEQC (
-        ch_mcl,
+    COGEQC(
+        ORTHOFINDER_MCL_TEST.out.inflation_dir,
         ch_annotations
     )
-    ch_summs = COGEQC.out.og_summary.collect()
+    ch_cogeqc_summary = COGEQC.out.cogeqc_summary.collect()
```

Comment:
again on the naming, should this be changed if we aren't using that library?

Reply:
See above